src/nvim/tui/tui.c

```diff
@@ -587,12 +608,21 @@ static void tui_set_mode(UI *ui, ModeShape mode)
                  data->params, out, ui, NULL, NULL);
   }
 
-  if (c.id != 0 && ui->rgb) {
-    int attr = syn_id2attr(c.id);
-    if (attr > 0) {
-      attrentry_T *aep = syn_cterm_attr2entry(attr);
-      data->params[0].i = aep->rgb_bg_color;
-      unibi_out(ui, data->unibi_ext.set_cursor_color);
+  // update cursor colors
+  if (c.id != 0) {
+    // int attr = syn_id2attr(c.id);
+    // attrentry_T *aep = syn_cterm_attr2entry(attr);
+    // cterm_bg_color
+    cursor_bg = c.attrs.background;
+
+
+    if (cursor_bg == -1) {
+      unibi_out(ui, unibi_cursor_invisible);
+    } else {
+      unibi_out(ui, unibi_cursor_normal);  // display if previously invisible
```

Comment:
It only happens on mode-change, so it's ok. Incidentally we do unibi_cursor_normal quite often, it's part of UI "flush".